GeneralsMD/Code/GameEngine/Source/GameClient/InGameUI.cpp

```diff
@@ -5899,3 +6024,82 @@ void InGameUI::drawGameTime()
 	m_gameTimeString->draw(horizontalTimerOffset, m_gameTimePosition.y, m_gameTimeColor, m_gameTimeDropColor);
 	m_gameTimeFrameString->draw(horizontalFrameOffset, m_gameTimePosition.y, GameMakeColor(180,180,180,255), m_gameTimeDropColor);
 }
+
+void InGameUI::drawLatencyCounter()
+{
+    const UnsignedInt latencyFrames = TheNetwork->getRunAhead();
+
+    UnicodeString latencyStr;
+		latencyStr.format(L"%u", latencyFrames);
+    m_latencyString->setText(latencyStr);
+
+    if (isAutoPos(m_latencyPosition))
+    {
+        m_latencyString->draw(kHudAnchorX, kHudAnchorY, m_latencyColor, m_latencyDropColor);
+    }
+    else
+    {
+        m_latencyString->draw(m_latencyPosition.x, m_latencyPosition.y, m_latencyColor, m_latencyDropColor);
+    }
+}
+
+void InGameUI::drawFPSCounter()
+{
+	const UnsignedInt nowMs = timeGetTime();
+	if (m_renderFpsLastUpdateMs == 0)
+	{
+		m_renderFpsLastUpdateMs = nowMs;
+		m_renderFpsFrameCount = 0;
+	}
+	m_renderFpsFrameCount += 1;
+	const UnsignedInt updateIntervalMs = 300;
+	if (nowMs - m_renderFpsLastUpdateMs >= updateIntervalMs)
+	{
+		UnsignedInt elapsed = nowMs - m_renderFpsLastUpdateMs;
+		if (elapsed == 0)
+			elapsed = 1;
+		const Real countedFps = (Real)m_renderFpsFrameCount * 1000.0f / (Real)elapsed;
+		m_renderFpsLastUpdateMs = nowMs;
+		m_renderFpsFrameCount = 0;
+		m_renderFpsDisplay = (Int)(countedFps + 0.5f);
+	}
+	Int renderFps = m_renderFpsDisplay;
+
+	UnsignedInt fpsLimit = 0u;
+	if (TheGlobalData->m_useFpsLimit && TheGameEngine)
+	{
+		fpsLimit = (UnsignedInt)TheGameEngine->getFramesPerSecondLimit();
+		if (fpsLimit == RenderFpsPreset::UncappedFpsValue)
+			fpsLimit = 0u;
+	}
+	if (fpsLimit > 0 && (renderFps >= (Int)fpsLimit - 1) && (renderFps <= (Int)fpsLimit + 1))
+		renderFps = (Int)fpsLimit;
+
+	UnicodeString fpsStr;
+	fpsStr.format(L"%d", renderFps);
+	m_renderFpsString->setText(fpsStr);
+	UnicodeString fpsLimitStr;
+	fpsLimitStr.format(L"[%u]", fpsLimit);
+	m_fpsLimitString->setText(fpsLimitStr);
+
+    const Bool renderFpsAuto = isAutoPos(m_renderFpsPosition);
+    if (renderFpsAuto)
+    {
+        Int x = kHudAnchorX;
+        // TheSuperHackers @info if latency is drawn at the same hud anchor, fps text is offset to render after it
+        const Bool latencyDrawn = (m_latencyPointSize > 0) && (TheGameLogic->isInMultiplayerGame() || TheGameLogic->isInLanGame() || TheGameLogic->isInInternetGame());
```

Comment:
This is not a good way to use offsets. Pass offsets from one function another like so:
```
int x = 0;
int y = 0;
drawFirstThing(x,y);
drawSecondThing(x,y);
```